lib/DidKeyDriver.js

```diff
@@ -140,11 +140,29 @@ export class DidKeyDriver {
 
   /**
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
+   * Note that unlike `generate()`, a `keyPairs` map is not returned. Use
+   * `publicMethodFor()` to fetch keys for particular proof purposes.
    *
    * @param {object} options - Options hashmap.
    * @typedef LDKeyPair
-   * @param {LDKeyPair} options.keyPair - A verification key pair to use to
-   *   generate the DID Document.
+   * @param {LDKeyPair|object} options.keyPair - Key used to generate the DID
+   *   document (either an LDKeyPair instance containing public key material,
+   *   or a "key description" plain object (such as that generated from a KMS)).
+   *
+   * @returns {Promise<object>} Resolves with the generated DID Document.
+   */
+  async publicKeyToDidDoc({keyPair} = {}) {
```

Comment:
The function signature invites developers to use poor hygiene when dealing with key material. The function name is `publicKeyToDidDoc` which is fine... but it takes the parameter `keyPair` which suggests that the function requires the public/private key pair. Though the jsdoc nudges the developer to use the public key material, I suggest more explicitness for our Public API.

```suggestion
  async publicKeyToDidDoc({publicKeyDescription} = {}) {
```

Reply:
@gannan08 - sure thing, happy to change the signature to your suggestion.

Reply:
@gannan08 - renamed.